core/src/main/java/org/apache/iceberg/BaseRowDelta.java

```diff
@@ -132,15 +144,44 @@ protected void validate(TableMetadata base, Snapshot parent) {
             parent);
       }
 
+      if (validateDeletes) {
+        failMissingDeletePaths();
+      }
+
       if (validateNewDataFiles) {
         validateAddedDataFiles(base, startingSnapshotId, conflictDetectionFilter, parent);
       }
 
       if (validateNewDeleteFiles) {
+        // validate that explicitly deleted files have not had added deletes
+        if (!deletedDataFiles.isEmpty()) {
```

Comment:
This block simply ensures no concurrent transactions added new DVs for files we are trying to remove?